backport/bulk_backport.py

```diff
@@ -65,44 +116,40 @@ def bulk_backport(
     )
     -- and must not come from ETL
     and sourceChecksum is null
+    -- and must not be archived
+    and not isArchived
     order by rand()
     limit %(limit)s
     """
 
-    # ignore limit if using dataset ids
-    if dataset_ids:
-        limit = 1000000
-
     df = pd.read_sql(
         q,
         engine,
         params={"limit": limit, "dataset_ids": dag_backported_ids},
     )
+    return cast(pd.DataFrame, df)
 
-    if dataset_ids:
-        df = df[df.id.isin(dataset_ids)]
 
-    df["short_name"] = df.name.map(underscore)
+def _prune_walden_datasets(engine: Engine, dry_run: bool) -> None:
+    active_dataset_ids = set(_active_datasets(engine)["id"])
 
-    log.info("bulk_backport.start", n=len(df))
+    walden_catalog = WaldenCatalog()
 
-    for i, ds in enumerate(df.itertuples()):
-        log.info(
-            "bulk_backport",
-            dataset_id=ds.id,
-            name=ds.name,
-            private=ds.isPrivate,
-            progress=f"{i + 1}/{len(df)}",
-        )
-        backport(
-            dataset_id=ds.id,
-            short_name=ds.short_name,
-            dry_run=dry_run,
-            upload=upload,
-            force=force,
-        )
+    datasets_to_delete = [
+        ds
+        for ds in walden_catalog.find(namespace="backport")
+        if utils.extract_id_from_short_name(ds.short_name) not in active_dataset_ids
+    ]
 
-    log.info("bulk_backport.finished")
+    log.info("bulk_backport.delete", n=len(datasets_to_delete))
+
+    for ds in datasets_to_delete:
+        log.info("bulk_backport.delete_dataset", short_name=ds.short_name)
+
+        # delete it from local and remote catalog
+        if not dry_run:
+            ds.delete()
+            ds.delete_from_remote()
```

Comment:
Should we make this delete_from_remote configurable of safeguard it? I'm not sure how often someone will run this but if you have walden write access configured and are testing something you might not be aware that prune deletes datasets not just in your local catalog but also the remote one. So maybe either add a --prune-remote flag or add a safety check for an env var that is by default only set on prod?

Reply:
Makes a lot of sense, better be safe than sorry. Added the `--prune-remote` flag.